### PR TITLE
[controller][server][dvc][vpj][producer][consumer] Fixed sys store init

### PR DESCRIPTION
**Pull request by FelixGV**
This fixes in an issue in NetworkClient::leastLoadedNode which manifested as a flood of IllegalArgumentException when some Kafka partition became completely unavailable.

## How was this PR tested?
It's about to be.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by sixpluszero — APPROVED** [review]

**Comment by FelixGV** [comment]
Integ test 1000 failed multiple times... there might be a regression. Will need to investigate before this can be merged...

**Comment by FelixGV** [comment]
Integ test 1000 seems to be flaky independent of this change, but I still tried to mitigate the flakiness in my latest commit... it's not very clean, IMHO, but it apparently did the trick in the one run after my commit went in...

**Comment by FelixGV** [comment]
@sixpluszero I added some unrelated mitigations to a flaky integration test, but the main change is still the same, and checks pass now. Would you mind taking another look when you get a chance? Thanks a lot 🙏

File: services/venice-controller/src/test/java/com/linkedin/venice/controller/init/SystemStoreInitializationHelperTest.java [hunk 1]
```diff
@@ -1,9 +1,7 @@
 package com.linkedin.venice.controller.init;
 
 import static com.linkedin.venice.controller.init.SystemStoreInitializationHelper.DEFAULT_KEY_SCHEMA_STR;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyInt;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
[the remark this answers is not in the document]
Comment (FelixGV) [reply services/venice-controller/src/test/java/com/linkedin/venice/controller/init/SystemStoreInitializationHelperTest.java]: Thanks for catching... done.

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/DaVinciClientMemoryLimitTest.java [hunk 2]
```diff
@@ -142,122 +159,168 @@ private VeniceProperties getDaVinciBackendConfig(
...
+    this.veniceCluster.createPushStatusSystemStore(store);
+
+    // Make sure system stores are enabled
+    this.veniceCluster.useControllerClient(cc -> {
+      StoreResponse storeResponse = TestUtils.assertCommand(cc.getStore(store));
       assertTrue(storeResponse.getStore().isDaVinciPushStatusStoreEnabled());
-      prepareMetaSystemStore(storeName);
-      // Do an VPJ push
-      runVPJ(vpjProperties, 1, controllerClient);
+      assertTrue(storeResponse.getStore().isStoreMetaSystemStoreEnabled());
```
Comment (sixpluszero) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/DaVinciClientMemoryLimitTest.java:197]: Just curious - isn't preparemetaSystemStore call equal to check meta store flag enabled? Or could you explain a bit what's the change that deflake the test?

**Review by nisargthakkar — APPROVED** [review]
LGTM

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/DaVinciClientMemoryLimitTest.java (hunk 2 above)
Comment (sixpluszero) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/DaVinciClientMemoryLimitTest.java:197]: I see, thank you for the explanation!

**Review by sixpluszero — APPROVED** [review]

**Pull request merged**